File: 3.0/golang/25.go
package main

import (
	"fmt"
	"math"
	"sort"
)

func main() {
	var N int
	fmt.Scanln(&N)

	var x int
	g := make([]int, N)
	for i := range g {
		fmt.Scan(&x)
		g[i] = x
	}
	sort.Ints(g)

	dp := make([]int, N+1)
	dp[2] = g[1] - g[0]

	if N > 2 {
		dp[3] = g[2] - g[0]
		for i := 4; i < N+1; i++ {
			dp[i] = int(math.Min(float64(dp[i-1]), float64(dp[i-2]))) + g[i-1] - g[i-2]
		}
	}

	fmt.Println(dp[N])
}

File: 3.0/python/25.py
n = int(input())
g = list(map(int, input().split()))

g.sort()

dp = [0] * (n + 1)
dp[2] = g[1] - g[0]

if n > 2:
    dp[3] = g[2] - g[0]
    for i in range(4, n + 1):
        dp[i] = min(dp[i - 1], dp[i - 2]) + g[i - 1] - g[i - 2]
print(dp[n])